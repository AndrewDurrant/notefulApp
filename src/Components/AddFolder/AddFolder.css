.addFolder section {
  margin: 1rem 0;
}

.addFolder label {
  display: block;
  text-transform: uppercase;
  font-weight: bold;
  margin-bottom: .5rem;
}

#folderName {
  font-size: 16px;
  width: 420px;
  padding: 5px;
}
