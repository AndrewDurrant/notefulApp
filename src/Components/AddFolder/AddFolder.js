import React, { Component } from 'react';
import cuid from 'cuid';
import ValidationError from '../../ValidationError';
import NotefulContext from '../../NotefulContext';




export class AddFolder extends Component {
  static contextType = NotefulContext;

  constructor(props) {
    super(props);
    this.state = {
      folderName: {
        value: '',
        touched: false
      }
    }
  }

  updateFolderName(name) {
    this.setState({
      folderName: {
        value: name, 
        touched: true
      }
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const { folderName } = this.state;
    const folder = {
      id: cuid(),
      name: folderName.value
    }
    console.log('Folder Name: ', folderName.value);
    this.context.addFolder(folder)
      .then(() => {
        // add folder
        this.context.updateFolders(folder)
        this.props.history.push('/')
      })
  }

  validateFolderName() {
    const folderName = this.state.folderName.value.trim();
    if(folderName.length === 0) {
      return 'Folder name is required';
    } else if(folderName.length < 3) {
      return 'Folder name must be at least 3 characters long'
    }
  }



  render() {
    // let context = this.context
    const folderNameError = this.validateFolderName();

    return (
      <form className="addFolder" onSubmit={e => this.handleSubmit(e)}>
        <h2>Add Folder</h2>
        <div>* required field</div>
        <div className="form-group">
          <label htmlFor="folderName">Folder Name *</label>
          <input 
            type="text"
            name="folderName"
            id="folderName"
            onChange={e => this.updateFolderName(e.target.value)}
          />
          {this.state.folderName.touched && (
            <ValidationError message={folderNameError} />
          )}
        </div>
        <div className="addFolder__button__group">

            <button type="reset"> 
              Cancel
            </button>
            <button
              type="submit"
              disabled= {
                this.validateFolderName()
              }
            >
              Save
            </button>
        </div>
        
      </form>
    )
  }
}

export default AddFolder
