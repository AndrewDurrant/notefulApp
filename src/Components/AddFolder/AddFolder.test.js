import React from 'react';
import ReactDOM from 'react-dom';
import AddFolder from './AddFolder';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';



it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(
    <MemoryRouter>
      <AddFolder />
    </MemoryRouter>
    , div
  );

  ReactDOM.unmountComponentAtNode(div);
})

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<MemoryRouter><AddFolder /></MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})