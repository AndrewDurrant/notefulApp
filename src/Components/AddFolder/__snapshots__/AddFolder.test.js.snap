// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  className="addFolder"
  onSubmit={[Function]}
>
  <h2>
    Add Folder
  </h2>
  <div>
    * required field
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="folderName"
    >
      Folder Name *
    </label>
    <input
      id="folderName"
      name="folderName"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="addFolder__button__group"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        type="reset"
      >
        Cancel
      </button>
    </a>
    <button
      disabled="Folder name is required"
      type="submit"
    >
      Save
    </button>
  </div>
</form>
`;
